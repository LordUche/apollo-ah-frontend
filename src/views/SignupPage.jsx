import React from 'react';
import { Link} from 'react-router-dom';
import SignupForm from '../components/SignupForm';

/**
 *This represents the actual signup page on the web
 *@class SignupPage
 *@extends {React.Component}
 */
class SignupPage extends React.Component {
  /**
   * 
   * @returns {JSX} - Signup page
   */
  render() {
    return (
      <main className="main-body">
        <section className="wrapper min-vh-100 ">
          <div className="mt-3 mb-3 container">
            <div className="signup-row">
              <div className="row">
                <div className="position-relative col-12 col-md-6 col-sm-12">
                  <Link
                    to="/"
                    className="signup-btn-back d-inline-block transition"
                  >
                    <i className="fas fa-arrow-left" />
                  </Link>
                  <SignupForm />
                </div>
                <div className="signup-bg relative overlay col-md-6">
                  <div className="signup-intro">
                    <h2>Express your ideas</h2>
                    <p className="font-weight-lighter">
                      Create an account to start reading and writing
                      beautiful articles.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </main>
    );
  }
}

export default SignupPage;