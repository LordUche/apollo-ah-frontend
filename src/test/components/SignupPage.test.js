import React from 'react';
import { shallow } from 'enzyme';
import SignupPage from '../../views/SignupPage.jsx';
const props = [{
  message: {message: "Email is required"},
  close: () => {
    return;
  }
}];

describe('<SignupPage Test Suite>', () => {
  describe('<SignupPage>', () => {
    it('It should render succesfully', () => {
      const wrapper = shallow(<SignupPage />);
      expect(wrapper).toBeDefined();
      console.log(wrapper.debug());
      expect(wrapper).toMatchSnapshot();
    });
  });
});