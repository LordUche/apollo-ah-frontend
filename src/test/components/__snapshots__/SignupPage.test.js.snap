// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupPage Test Suite> <SignupPage> It should render succesfully 1`] = `ShallowWrapper {}`;
